### PR TITLE
Add support to create and preview SQL files

### PR DESCRIPTION
**Pull request by javierluraschi**
Similar to D3 files, this PR adds support to create and preview SQL files given a connection and a script:

screen shot 2018-04-30 at 6 16 37 pm

SQL files become preview-able with

```
-- !preview dbGetQuery conn=<connection> <parameters>
```

For instance and by default, we can use a SQLite connection as follows:

```
-- !preview dbGetQuery conn=DBI::dbConnect(RSQLite::SQLite())
```

screen shot 2018-04-30 at 6 22 32 pm

Which on preview, displays as a console tab:

screen shot 2018-04-30 at 10 59 40 pm

### CODE REVIEW

File: src/cpp/session/modules/SessionDataPreview.R [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+# AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+#
+#
+
+.rs.addFunction("previewDataFrame", function(data, script)
+{
+   max <- 100
+
+   columns <- list()
```
Comment (jmcphers) [inline src/cpp/session/modules/SessionDataPreview.R]: should this share an implementation with https://github.com/rstudio/rstudio/blob/560aeb25f6634b3187082280e192d595ebc047be/src/cpp/session/modules/SessionDataImportV2.R#L767-L774?

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetWidget.java [hunk 2]
```diff
@@ -399,9 +399,12 @@ public void onClick(ClickEvent event)
       sourceButton_.setTitle(SOURCE_BUTTON_TITLE);
       toolbar.addRightWidget(sourceButton_);
 
-      previewButton_ = commands_.previewJS().createToolbarButton(false);
-      toolbar.addRightWidget(previewButton_);
+      previewJsButton_ = commands_.previewJS().createToolbarButton(false);
+      toolbar.addRightWidget(previewJsButton_);
       
+      previewSqlButton_ = commands_.previewSql().createToolbarButton(false);
```
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetWidget.java]: this probably needs an icon that doesn't have the globe on it (which implies HTML) -- maybe the green preview arrow over an icon of a data grid? (we can get this from paul after merge if you like)

File: src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml [hunk 3]
```diff
@@ -894,6 +897,11 @@ well as menu structures (for main menu and popup menus).
         menuLabel="R Doc_umentation"
         desc="Create a new Rd documentation file"
         rebindable="false"/>
+
+   <cmd id="newSqlDoc"
+        menuLabel="S_ql Script"
```
Comment (hadley) [inline src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml]: `S_QL`?

File: src/cpp/session/modules/SessionDataPreview.R (hunk 1 above)
Comment (javierluraschi) [reply src/cpp/session/modules/SessionDataPreview.R]: Can do.

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetWidget.java (hunk 2 above)
Comment (javierluraschi) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetWidget.java]: Yeah, makes sense, I'll ask Paul for icons on SQL preview and D3 preview, but could be refined a bit.

File: src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml (hunk 3 above)
Comment (javierluraschi) [reply src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml]: Much better, thank you!

**Comment by javierluraschi** [comment]
See also https://github.com/rstudio/rstudioapi/pull/96.

**Comment by jjallaire** [comment]
Instead of "Source on Save" it should be "Preview on Save" (as we do for D3 and Rd files).

**Pull request merged**